AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs

```diff
@@ -53,28 +54,52 @@ public void DeleteMachine()
         }
 
         public void DeployAllCurrentReleasesToThisMachine()
-        {
+        {            
+            const string targetRolePropertyName = "Octopus.Action.TargetRoles";
             List<TaskState> currentStatuses;
             try
             {
                 var machineId = _repository.Machines.FindByName(_machineName).Id;
                 var environment = _repository.Environments.FindByName(_config.TentacleEnvironment).Id;
+                var projects = _repository.Projects.FindAll()

```

Comment:
My assumption was that this would make one call to get the projects, then one call per project to get the deployment process for that project. I have not watched the traffic to test this however. I had problems running locally so have only tested "in production".
I wasn't too concerned about the number of API calls because this is only for tentacle start-up.